org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/ClosedProjectsQueue.java

```diff
@@ -59,7 +59,12 @@ public boolean isEmpty() {
 		 * Add the tasks again to the top of the queue.
 		 */
 		public void reschedule() {
-			insert(projectNames, toBeBuilt);
+			Set<URI> toBeDeleted = toBeBuilt.getToBeDeleted();
```

Comment:
is this executed single threaded

Reply:
Yes, it's running in the builder with a lock.